viz/plant_status/presentation.py

```diff
@@ -0,0 +1,95 @@
+from enum import StrEnum
+from pydantic import BaseModel
+from decimal import Decimal
+from datetime import datetime
+from etl.models import Plant
+from viz.plant_status.status import PlantStatusSolver
+
+from viz.plant_status.data_models import (
+    BalancingDirection,
+)
+
+
+class Card(BaseModel): ...
+
+
+class BadgeClass(StrEnum):
+    PRIMARY = "text-bg-primary"
+    SECONDARY = "text-bg-secondary"
+    SUCCESS = "text-bg-success"
+    INFO = "text-bg-info"
+    WARNING = "text-bg-warning"
+    DANGER = "text-bg-danger"
+
+
+class StatusClass(StrEnum):
+    GENERATING = "status-generating"
+    STANDBY = "status-standby"
+    BALANCING = "status-balancing"
+    ACCEPTED_BID = "status-accepted-bid"
+    TRIP = "status-trip"
+    OUTAGE = "status-outage"
+
+
+class StatusText(StrEnum):
+    GENERATING = "Generating"
+    STANDBY = "Standby"
+    BALANCING = "Balancing"
+    ACCEPTED_BID = "Accepted Bid"
+    TRIP = "Trip"
+    OUTAGE = "Outage"
+
+
+class PlantStatusCard(Card):
+    name: str
+    status_class: StatusClass
+    badge_class: BadgeClass
+    status_text: StatusText
+    icon: str | None
+    actual_gen: Decimal
```

Comment:
You're right to use Decimals for these. Just keep in mind DB / timescale models use FloatFields, so conversion would have to be done with some care.